packages/expect/src/jest-extend.ts

```diff
@@ -108,6 +108,13 @@ function JestExtendPlugin(expect: ExpectStatic, matchers: MatchersObject): ChaiP
         }
       }
 
+      Object.defineProperty(((globalThis as any).__hackAsymmetricMatchers), expectAssertionName, {
```

Comment:
I'm fine with this approach, but we should use Symbols here - put it into `src/constants.ts`